wibotic_connector_can/src/wibotic_connector_can/wiboticros.py

```diff
@@ -46,136 +47,91 @@ class Shutdown(Exception):
-                uavcan_dsdl_type = uavcan.get_uavcan_data_type(
-                    incoming_data
-                )
+                uavcan_dsdl_type = uavcan.get_uavcan_data_type(incoming_data)
                 unpacked_data = {}
                 for field in uavcan_dsdl_type.fields:
-                    unpacked_data[field.name] = getattr(
-                        incoming_data, field.name
-                    )
-                packaged_data = msg.WiBoticInfo(
-                    **unpacked_data
-                )
+                    unpacked_data[field.name] = getattr(incoming_data, field.name)
+                packaged_data = msg.WiBoticInfo(**unpacked_data)
                 rospy.loginfo(packaged_data)
                 pub.publish(packaged_data)
 
-        def handle_param_list(self, req):
-            params = []
-            index = 0
-            ClearQueue(_uav_incoming_param)
-            while True:
-                request = uavcan.protocol.param.GetSet.Request(
-                    index=index
-                )
-                _uav_outgoing.put(request)
-                try:
-                    response = _uav_incoming_param.get(
-                        block=True,
-                        timeout=PARAMETER_REQ_TIMEOUT,
-                    )
-                    if (
-                        uavcan.get_active_union_field(
-                            response.value
-                        )
-                        != "empty"
-                    ):
-                        params.append(
-                            ascii_list_to_str(response.name)
-                        )
-                        index += 1
-                        rospy.loginfo(index)
-                    else:
-                        return [SUCCESS, params]
-                except queue.Empty:
-                    return [FAILURE, params]
-
-        def handle_param_read(self, req):
-            request = uavcan.protocol.param.GetSet.Request(
-                name=req.name
-            )
+        @staticmethod
+        def request_param(request):
             _uav_outgoing.put(request)
             try:
                 response = _uav_incoming_param.get(
                     block=True,
                     timeout=PARAMETER_REQ_TIMEOUT,
                 )
-                if (
-                    uavcan.get_active_union_field(
-                        response.value
-                    )
-                    != "empty"
-                ):
-                    status = (
-                        SUCCESS
-                        if response.name == req.name
-                        else FAILURE
-                    )
-                    value = response.value.integer_value
-                    return [status, value]
             except queue.Empty:
-                pass
+                return FAILURE
 
+            if uavcan.get_active_union_field(response.value) == "empty":
+                return "empty"
+            return response
```

Comment:
Parameter Handling:

Factored this out of the other 4 `handle_param_xxxxx()` methods to increase clarity and prevent issues with future changes.